### ISSUE TITLE
[ RFC ] API Endpoint Design for Filtering Progresses by User and Date

### ISSUE THREAD

**Issue by heyrandhir** [Real-Dev-Squad/website-backend#1105]
**Abstract**

This RFC aims to discuss the appropriate API endpoint design for filtering progresses based on user and date parameters in a RESTful API. The goal is to determine the most suitable approach for providing filtering capabilities while adhering to RESTful principles and maintaining API consistency.

When implementing an API endpoint to retrieve progresses based on user and date criteria, two common approaches are often considered:

Using Path Parameters: `/progresses/user/XGqmn2yJoneT1aXXh22P/date/2023-05-21`
Using Query Parameters: `/progresses?userId=XGqmn2yJoneT1aXXh22P&date=2023-05-21`

This RFC aims to discuss the advantages and considerations of each approach to determine the most RESTful and effective solution.

**_Path Parameters_**
Using path parameters in the API endpoint structure can provide a more hierarchical and resource-oriented design. It is commonly used when the values are considered mandatory and represent essential parts of the resource identification. However, it's important to note that path parameters are not typically used for filtering data, but rather for specifying specific resource identifiers.

_Query Parameters_
Using query parameters offers a more flexible and concise approach to provide filtering capabilities. Query parameters are commonly used when the values are optional and used for filtering and pagination. Query parameters allow for a clear separation between the base endpoint and the filtering parameters, making the API more intuitive for API consumers.

**Comment by heyrandhir** [comment]
As per my understanding , We should go with 
```js
/progresses/user/XGqmn2yJoneT1aXXh22P/date/2023-05-21
```
This approach ensures clarity and provides a clear indication of the specific resource being accessed. By including the date parameter in the path, we eliminate any ambiguity and clearly communicate its importance in identifying the requested progress document.

**Comment by shubham-y** [comment]
@heyrandhir Will the combination of date and user ID values be unique for each document?

**Comment by heyrandhir** [comment]
Yes @shubham-y for a given date and user Id there will be a unique document.
